forge-gui/res/cardsfolder/upcoming/queen_kayla_bin-kroog.txt

```diff
@@ -0,0 +1,15 @@
+Name:Queen Kayla bin-Kroog
+ManaCost:1 R W
+Types:Legendary Creature Human Noble
+PT:2/3
+A:AB$ Discard | Cost$ 4 T | Defined$ You | RememberDiscarded$ True | Mode$ Hand | SubAbility$ DBDraw | SorcerySpeed$ True | SpellDescription$ Discard all the cards in your hand, then draw that many cards. You may choose an artifact or creature card with mana value 1 you discarded this way, then do the same for artifact or creature cards with mana values 2 and 3. Return those cards to the battlefield. Activate only as a sorcery.
+SVar:DBDraw:DB$ Draw | Defined$ You | NumCards$ X | SubAbility$ DBChooseCardOne
+SVar:DBChooseCardOne:DB$ ChooseCard | Defined$ You | Choices$ Artifact.IsRemembered+cmcEQ1,Creature.IsRemembered+cmcEQ1 | ImprintChosen$ True | ChoiceTitle$ You may choose an artifact or creature card with mana value 1 you discarded this way | ChoiceZone$ Graveyard | AILogic$ BestCard | SubAbility$ DBChooseCardTwo 
+SVar:DBChooseCardTwo:DB$ ChooseCard | Defined$ You | Choices$ Artifact.IsRemembered+cmcEQ2,Creature.IsRemembered+cmcEQ2 | ImprintChosen$ True  | ChoiceTitle$ You may choose an artifact or creature card with mana value 2 you discarded this way | ChoiceZone$ Graveyard | AILogic$ BestCard | SubAbility$ DBChooseCardThree
+SVar:DBChooseCardThree:DB$ ChooseCard | Defined$ You | Choices$ Artifact.IsRemembered+cmcEQ3,Creature.IsRemembered+cmcEQ3 | ImprintChosen$ True | ChoiceTitle$ You may choose an artifact or creature card with mana value 3 you discarded this way | ChoiceZone$ Graveyard | AILogic$ BestCard | SubAbility$ DBChangeZoneAll
+SVar:DBChangeZoneAll:DB$ ChangeZoneAll | Origin$ Graveyard | ChangeType$ Card.IsImprinted | Destination$ Battlefield | SubAbility$ DBCleanup
```

Comment:
@tool4ever @Hanmac is it more efficient to do it this way or with `DB$ ChangeZone | Defined$ Imprinted` ?

Reply:
PS nice work @Simisays 💪

Reply:
`Defined$ Imprinted` is cleaner. Assuming the cards are in a public zone (graveyard or exile?)